components/econet/econet.cpp

```diff
@@ -52,27 +52,15 @@ void address_to_bytes(uint32_t adr, std::vector<uint8_t> *data) {
   data->push_back(0);
 }
 
-// Extracts strings in pdata separated by 0x00
+// Extracts strings of length OBJ_NAME_SIZE in pdata separated by 0x00, 0x00
 void extract_obj_names(const uint8_t *pdata, uint8_t data_len, std::vector<std::string> *obj_names) {
   const uint8_t *start = pdata + 4;
-  while (true) {
-    // Look for the first occurrence of 0x00 in the remaining bytes
-    size_t num = data_len - (start - pdata);
-    const uint8_t *end = (const uint8_t *) memchr(start, 0, num);
-    if (!end) {
-      // Not found, so add all the remaining bytes and finish
-      std::string s((const char *) start, num);
-      obj_names->push_back(s);
-      break;
-    }
-    // Add all bytes until the first occurrence of 0x00
-    std::string s((const char *) start, end - start);
+  const uint8_t *endp = pdata + data_len;
+  while (start < endp) {
+    const uint8_t *end = std::min(start + OBJ_NAME_SIZE, endp);
```

Comment:
I was just going to make a comment that we shouldn't always assume that OBJ_NAME_SIZE will be 8... but even in cases where the actual object name size is less than 8 such as "SPT", it fills the space with trailing 0x00's. So as far as I know this is a safe assumption we can make. We should keep an eye out for this.